# CLEANING THE DATAFRAME #

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [267]:
df = pd.read_csv('../data/final_properties_dataset_260304.csv', sep = ',')


## Analyzing the dataset ##

In [268]:
df.shape

(13195, 20)

In [203]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 13195 entries, 0 to 13194
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   URL                     13195 non-null  str    
 1   Zip code                13171 non-null  float64
 2   City                    13171 non-null  str    
 3   Type of property        13195 non-null  str    
 4   Subtype of property     13195 non-null  str    
 5   Price                   12973 non-null  float64
 6   Type of sale            13195 non-null  str    
 7   Number of rooms         12925 non-null  float64
 8   Living Area             12352 non-null  float64
 9   Fully equipped kitchen  4586 non-null   float64
 10  Furnished               8272 non-null   float64
 11  Open fire               1168 non-null   float64
 12  Terrace                 12548 non-null  float64
 13  Terrace area            5692 non-null   float64
 14  Garden                  12114 non-null  float64
 

In [204]:
df.head()

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,NaN,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,NaN,NaN,NaN,1.0,NaN,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,NaN,0.0,1.0,NaN,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,NaN,1.0,NaN,1.0,NaN,2647.0,4.0,1.0,2.0


In [205]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,13195,13171.000000,13171,13195,13195,1.297300e+04,13195,12925.000000,12352.000000,4586.000000,8272.000000,1168.000000,12548.000000,5692.000000,12114.000000,3006.000000,6195.000000,10300.000000,4487.000000,10379.000000
unique,13195,NaN,1420,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,470,7014,5911,NaN,13077,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3871.155721,NaN,NaN,NaN,4.361525e+05,NaN,3.100812,164.565981,2.232883,0.048477,0.309075,0.780762,25.150914,0.591795,1130.354291,1252.880549,2.748350,0.090038,1.615666
std,NaN,2851.772144,NaN,NaN,NaN,5.259908e+05,NaN,17.336015,136.217282,0.705065,0.214784,0.462310,0.413747,52.496356,0.491522,13158.124785,3893.797538,0.845458,0.286268,1.400379
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,91.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,84.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3090.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,133.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,230.000000,497.000000,2.000000,0.000000,1.000000
75%,NaN,6220.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,195.000000,3.000000,0.000000,1.000000,1.000000,30.000000,1.000000,673.750000,1040.000000,4.000000,0.000000,2.000000


In [269]:
df.corr()

ValueError: could not convert string to float: 'https://immovlan.be/en/detail/residence/for-sale/9300/aalst/rbt64159'

### COLUMN: Price ###

In [206]:
# checking suspicious no price
def func_no_price() -> list:
    no_price_list = df[df['Price'].isnull()].index.tolist()
    print(f'There is(are) {len(no_price_list)} property(ies) without price. List of indices: {no_price_list}\n')
    for idx in no_price_list:
        print(f'No price is provided for the line: {idx}, URL: {df.loc[idx, 'URL']}')

In [207]:
func_no_price()

There is(are) 222 property(ies) without price. List of indices: [8, 43, 44, 110, 257, 274, 312, 344, 438, 506, 515, 617, 620, 642, 673, 690, 782, 795, 1011, 1148, 1283, 1504, 1588, 1608, 1684, 1694, 1706, 1722, 1724, 1795, 1871, 1901, 1921, 2137, 2150, 2179, 2189, 2225, 2280, 2358, 2422, 2453, 2463, 2485, 2518, 2529, 2584, 2677, 2754, 2788, 2816, 2875, 2902, 3018, 3086, 3133, 3234, 3300, 3339, 3375, 3378, 3498, 3524, 3525, 3606, 3618, 3674, 3690, 3699, 3728, 3755, 3768, 3806, 3822, 3823, 3899, 3989, 4030, 4156, 4209, 4364, 4383, 4416, 4447, 4471, 4528, 4586, 4602, 4609, 4625, 4691, 4721, 4843, 4869, 4898, 5073, 5077, 5137, 5275, 5293, 5317, 5329, 5359, 5380, 5525, 5600, 5949, 5959, 5977, 5996, 6102, 6215, 6239, 6264, 6292, 6436, 6525, 6636, 6730, 6774, 6782, 6871, 6936, 7064, 7069, 7079, 7083, 7090, 7147, 7154, 7163, 7168, 7367, 7422, 7561, 7569, 7583, 7594, 7607, 7723, 7760, 7860, 7877, 7926, 8003, 8012, 8017, 8155, 8164, 8232, 8237, 8242, 8321, 8368, 8374, 8386, 8396, 8638, 8697, 870

In [208]:
df.dropna(subset=['Price'], inplace = True)

In [209]:
df.info()

<class 'pandas.DataFrame'>
Index: 12973 entries, 0 to 13194
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   URL                     12973 non-null  str    
 1   Zip code                12973 non-null  float64
 2   City                    12973 non-null  str    
 3   Type of property        12973 non-null  str    
 4   Subtype of property     12973 non-null  str    
 5   Price                   12973 non-null  float64
 6   Type of sale            12973 non-null  str    
 7   Number of rooms         12729 non-null  float64
 8   Living Area             12181 non-null  float64
 9   Fully equipped kitchen  4536 non-null   float64
 10  Furnished               8155 non-null   float64
 11  Open fire               1164 non-null   float64
 12  Terrace                 12352 non-null  float64
 13  Terrace area            5617 non-null   float64
 14  Garden                  11921 non-null  float64
 15  G

In [210]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,4536.000000,8155.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,3.100558,163.649782,2.233245,0.049050,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,17.467203,134.782079,0.705816,0.215985,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,3.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000


### COLUMN: Zip code ###

In [211]:
# checking susoicious no zip code
def func_no_zip():
    no_zip_list = df[df['Zip code'].isnull()].index.tolist()
    print(f"There are {len(no_zip_list)} properties without zip\n")
    print(f'Indices of the property without zip code are: {no_zip_list}')
    for idx in no_zip_list:
        print(f'No index is provided for the line: {idx}, URL: {df.loc[idx, 'URL']}')

In [212]:
func_no_zip()

There are 0 properties without zip

Indices of the property without zip code are: []


### COLUMN: City ###

### COLUMN: Number of rooms ###

In [213]:
# checking susoicious max room number
def func_max_rooms():
    max_rooms = df['Number of rooms'].idxmax()
    print(f'index of the property with maximum number of rooms is: {max_rooms}, data: {df.loc[max_rooms, 'Number of rooms']}')

    print(f"{df.loc[max_rooms, 'URL']}")

In [214]:
func_max_rooms()

index of the property with maximum number of rooms is: 3499, data: 1962.0
https://immovlan.be/en/detail/residence/for-sale/1880/nieuwenrode/rbv11758


In [215]:
df.loc[3499, 'Number of rooms'] = 4

In [216]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,4536.000000,8155.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.233245,0.049050,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.705816,0.215985,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,3.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000


### COLUMN: Living area ###

In [217]:
# checking suspicious min living area number
def func_min_living_area():
    min_living_area = df['Living Area'].idxmin()
    print(f'index of the property with minimum living area is: {min_living_area}, data: {df.loc[min_living_area, 'Living Area']}')

    print(f"{df.loc[min_living_area, 'URL']}")

In [218]:
func_min_living_area()

index of the property with minimum living area is: 192, data: 1.0
https://immovlan.be/en/detail/residence/for-sale/4821/andrimont/vwd16705


### COLUMN: Fully equiped kitchen ###

In [219]:
n_fully_equiped_kitchen_missing = df['Fully equipped kitchen'].isnull().sum()
print(f"Number of missing data on fully equiped kitchen: {n_fully_equiped_kitchen_missing}")

fully_equiped_kitchen_mean_old = df['Fully equipped kitchen'].mean()
print(f"The fully equiped kitchen 'mean' BEFORE fillna: {fully_equiped_kitchen_mean_old}")
fully_equiped_kitchen_mode = df['Fully equipped kitchen'].mode()[0]
print(f"The fully equiped kitchen 'mode': {fully_equiped_kitchen_mode}")
df.describe()

Number of missing data on fully equiped kitchen: 8437
The fully equiped kitchen 'mean' BEFORE fillna: 2.2332451499118164
The fully equiped kitchen 'mode': 2.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,4536.000000,8155.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.233245,0.049050,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.705816,0.215985,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,3.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [220]:
# df.fillna({"Calories": 130}, inplace=True)
df.fillna({'Fully equipped kitchen': fully_equiped_kitchen_mode}, inplace = True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,NaN,NaN,1.0,NaN,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,NaN,0.0,1.0,NaN,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,NaN,1.0,NaN,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,NaN,1.0,NaN,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,NaN,NaN,0.0,NaN,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,NaN,NaN,1.0,NaN,1.0,NaN,288.0,3.0,NaN,4.0


In [221]:
fully_equiped_kitchen_mean_new = df['Fully equipped kitchen'].mean()
print(f"The fully equiped kitchen 'mean' AFTER fillna: {fully_equiped_kitchen_mean_new}")

The fully equiped kitchen 'mean' AFTER fillna: 2.0815539967625067


### COLUMN: Furnished ###

In [222]:
n_furnished_missing = df['Furnished'].isnull().sum()
print(f"Number of missing data on furnished: {n_furnished_missing}")

furnished_mean_old = df['Furnished'].mean()
print(f"The furnished 'mean' BEFORE fillna: {furnished_mean_old}")
furnished_mode = df['Furnished'].mode()[0]
print(f"The furnished 'mode': {furnished_mode}")
df.describe()

Number of missing data on furnished: 4818
The furnished 'mean' BEFORE fillna: 0.049049662783568364
The furnished 'mode': 0.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,8155.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.049050,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.215985,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [223]:
df.fillna({'Furnished': 0}, inplace = True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,NaN,1.0,NaN,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,NaN,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,NaN,1.0,NaN,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,NaN,1.0,NaN,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,NaN,0.0,NaN,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,NaN,NaN,NaN,NaN,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,NaN,1.0,NaN,1.0,NaN,288.0,3.0,NaN,4.0


In [224]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,12973.000000,12973.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.081554,0.030833,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.431896,0.172872,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,2.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000


In [225]:
furnished_mean_new = df['Furnished'].mean()
print(f"The furnished 'mean' AFTER fillna: {furnished_mean_new}")

The furnished 'mean' AFTER fillna: 0.030833269097356047


### COLUMN: Open fire ###

In [226]:
n_open_fire_missing = df['Open fire'].isnull().sum()
print(f"Number of missing data on open fire: {n_open_fire_missing}")

open_fire_mean_old = df['Open fire'].mean()
print(f"The open fire 'mean' BEFORE fillna: {open_fire_mean_old}")
open_fire_mode = df['Open fire'].mode()[0]
print(f"The open fire 'mode': {open_fire_mode}")
df.describe()

Number of missing data on open fire: 11809
The open fire 'mean' BEFORE fillna: 0.3075601374570447
The open fire 'mode': 0.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,1164.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.307560,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.461682,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,1.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [227]:
df.fillna({'Open fire': 0}, inplace=True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,NaN,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,0.0,1.0,NaN,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,0.0,0.0,NaN,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,0.0,NaN,NaN,NaN,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,288.0,3.0,NaN,4.0


In [228]:
df.describe()

,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [229]:
open_fire_mean_new = df['Open fire'].mean()
print(f"The open fire 'mean' AFTER fillna: {open_fire_mean_new}")

The open fire 'mean' AFTER fillna: 0.02759577584213366


### COLUMN: Terrace ###

In [230]:
n_terrace_missing = df['Terrace'].isnull().sum()
print(f"Number of missing data on terrace: {n_terrace_missing}")

terrace_mean_old = df['Terrace'].mean()
print(f"The terrace 'mean' BEFORE fillna: {terrace_mean_old}")
terrace_mode = df['Terrace'].mode()[0]
print(f"The terrace 'mode': {terrace_mode}")
df.describe()

Number of missing data on terrace: 621
The terrace 'mean' BEFORE fillna: 0.7802784974093264
The terrace 'mode': 1.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12352.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.780278,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.414075,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,1.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [231]:
df.fillna({'Terrace': 0}, inplace=True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,NaN,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,0.0,1.0,NaN,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,0.0,0.0,NaN,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,0.0,0.0,NaN,NaN,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,0.0,1.0,NaN,1.0,NaN,288.0,3.0,NaN,4.0


In [232]:
terrace_mean_new = df['Terrace'].mean()
print(f"The terrace 'mean' AFTER fillna: {terrace_mean_new}")

df['Terrace'].value_counts()


The terrace 'mean' AFTER fillna: 0.742927618900794


Terrace
1.0    9638
0.0    3335
Name: count, dtype: int64

### COLUMN: Terrace area ###

In [233]:
df.describe(include='all')
# checking suspicious terrace area size
def func_max_terrace_area():
    max_terrace_area = df['Terrace area'].idxmax()
    print(f'Index of the property with maximum terrace area size is: {max_terrace_area}, data: {df.loc[max_terrace_area, 'Terrace area']}')

    print(f"{df.loc[max_terrace_area, 'URL']}")

In [234]:
func_max_terrace_area()

Index of the property with maximum terrace area size is: 1199, data: 2968.0
https://immovlan.be/en/detail/residence/for-sale/6950/harsin/vbd88226


In [235]:
n_terrace_area_missing = df['Terrace area'].isnull().sum()
print(f"Number of missing data on terrace area: {n_terrace_area_missing}")

terrace_area_mean_old = df['Terrace area'].mean()
print(f"The terrace area mean BEFORE fillna: {terrace_area_mean_old}")
terrace_area_mode = df['Terrace area'].mode()[0]
print(f"The terrace area mode: {terrace_area_mode}")
df.describe()

Number of missing data on terrace area: 7356
The terrace area mean BEFORE fillna: 25.02207584119637
The terrace area mode: 20.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,5617.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,25.022076,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,52.443382,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,9.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,16.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,30.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [236]:
df.loc[(df['Terrace area'].isna()) & (df['Terrace'] == 1), 'Terrace area'] = terrace_area_mode

In [237]:
df.describe()

,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,9638.000000,11921.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,22.926852,0.589883,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,40.110923,0.491875,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,14.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [238]:
df.fillna({'Terrace area': 0}, inplace=True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,20.0,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,0.0,1.0,20.0,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,0.0,0.0,0.0,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,0.0,0.0,0.0,NaN,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,288.0,3.0,NaN,4.0


In [239]:
terrace_area_mean_new = df['Terrace area'].mean()
print(f"The terrace area 'mean' AFTER fillna: {terrace_area_mean_new}")

The terrace area 'mean' AFTER fillna: 17.03299159793417


### COLUMN: Garden ###

In [240]:
n_garden_missing = df['Garden'].isnull().sum()
print(f"Number of missing data on garden: {n_garden_missing}")

garden_mean_old = df['Garden'].mean()
print(f"The garden 'mean' BEFORE fillna: {garden_mean_old}")
garden_mode = df['Garden'].mode()[0]
print(f"The garden 'mode': {garden_mode}")


Number of missing data on garden: 1052
The garden 'mean' BEFORE fillna: 0.5898833990437043
The garden 'mode': 1.0


In [241]:
df.fillna({'Garden': 0}, inplace=True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,20.0,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,0.0,1.0,20.0,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,0.0,0.0,0.0,1.0,NaN,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,0.0,0.0,0.0,0.0,NaN,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,288.0,3.0,NaN,4.0


In [242]:
garden_mean_new = df['Garden'].mean()
print(f"The garden 'mean' AFTER fillna: {garden_mean_new}")

df['Garden'].value_counts()

The garden 'mean' AFTER fillna: 0.5420488707315193


Garden
1.0    7032
0.0    5941
Name: count, dtype: int64

### COLUMN: Garden area ###

In [243]:
# checking suspicious garden size
def func_max_garden_area():
    max_garden = df['Garden area'].idxmax()
    print(f'index of the property with maximum garden area size is: {max_garden}, data: {df.loc[max_garden, 'Garden area']}')

    print(f"{df.loc[max_garden, 'URL']}")

In [244]:
func_max_garden_area()

index of the property with maximum garden area size is: 8745, data: 690330.0
https://immovlan.be/en/detail/residence/for-sale/1130/haren/vbd76660


In [245]:
n_garden_area_missing = df['Garden area'].isnull().sum()
print(f"Number of missing data on garden area: {n_garden_area_missing}")

garden_area_mean_old = df['Garden area'].mean()
print(f"The garden area mean BEFORE fillna: {garden_area_mean_old}")
garden_area_mode = df['Garden area'].mode()[0]
print(f"The garden area mode: {garden_area_mode}")
df.describe()

Number of missing data on garden area: 10021
The garden area mean BEFORE fillna: 1095.1006097560976
The garden area mode: 100.0


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,1095.100610,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,13185.590866,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,225.500000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,655.500000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,690330.000000,101895.000000,8.000000,1.000000,4.000000


In [246]:
func_max_garden_area()

index of the property with maximum garden area size is: 8745, data: 690330.0
https://immovlan.be/en/detail/residence/for-sale/1130/haren/vbd76660


In [247]:
df.loc[8745, 'Garden area'] = df.loc[8745, 'Garden area']/10000
# df.loc[8745, 'Garden area'] = 69
df.loc[8745, 'Garden area']


np.float64(69.033)

In [248]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,2952.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,861.272369,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,3581.608616,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,83.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,225.000000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,653.250000,1030.000000,4.000000,0.000000,2.000000


In [249]:
df.head()

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,20.0,1.0,NaN,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,20.0,1.0,NaN,2647.0,4.0,1.0,2.0


In [250]:
# df.loc[(df['ColumnA'].isna()) & (df['ColumnB'] == 'нужное_значение'),'ColumnA'] = 'новое_значение'

df.loc[(df['Garden area'].isna()) & (df['Garden'] == 1), 'Garden area'] = garden_area_mode

In [251]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,7032.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,419.578503,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,2350.578016,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,100.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,100.000000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,150.000000,1030.000000,4.000000,0.000000,2.000000


In [252]:
df.fillna({'Garden area': 0}, inplace=True)

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
0,https://immovlan.be/en/detail/residence/for-sa...,9300.0,Aalst,House,residence,495000.0,for sale,5.0,310.0,2.0,0.0,0.0,1.0,25.0,1.0,114.0,NaN,NaN,NaN,1.0
1,https://immovlan.be/en/detail/villa/for-sale/6...,6840.0,Grandvoir,House,villa,649000.0,for sale,4.0,188.0,2.0,0.0,0.0,1.0,20.0,1.0,100.0,1479.0,NaN,NaN,NaN
2,https://immovlan.be/en/detail/residence/for-sa...,7850.0,Enghien,House,residence,675000.0,for sale,6.0,416.0,3.0,0.0,1.0,1.0,100.0,1.0,100.0,665.0,3.0,0.0,1.0
3,https://immovlan.be/en/detail/villa/for-sale/5...,5360.0,Hamois,House,villa,349000.0,for sale,3.0,162.0,3.0,0.0,0.0,1.0,20.0,1.0,100.0,NaN,4.0,NaN,2.0
4,https://immovlan.be/en/detail/villa/for-sale/1...,1780.0,Wemmel,House,villa,1149000.0,for sale,5.0,396.0,2.0,0.0,0.0,1.0,20.0,1.0,100.0,2647.0,4.0,1.0,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13190,https://immovlan.be/en/detail/residence/for-sa...,1180.0,Ukkel,House,residence,1000000.0,for sale,7.0,350.0,2.0,0.0,0.0,1.0,20.0,1.0,150.0,300.0,3.0,NaN,0.0
13191,https://immovlan.be/en/detail/residence/for-sa...,3945.0,Ham,House,residence,415000.0,for sale,3.0,329.0,2.0,0.0,0.0,0.0,0.0,1.0,100.0,1895.0,2.0,NaN,NaN
13192,https://immovlan.be/en/detail/residence/for-sa...,9420.0,Erpe-Mere,House,residence,512495.0,for sale,4.0,206.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,935.0,3.0,NaN,4.0
13193,https://immovlan.be/en/detail/residence/for-sa...,1785.0,Merchtem,House,residence,469000.0,for sale,3.0,163.0,2.0,0.0,0.0,1.0,20.0,1.0,100.0,288.0,3.0,NaN,4.0


In [253]:
garden_area_mean_new = df['Garden area'].mean()
print(f"The garden area mean AFTER fillna: {garden_area_mean_new}")
df.describe()

The garden area mean AFTER fillna: 227.4320537269714


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,6086.000000,10172.000000,4415.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,227.432054,1210.621098,2.744691,0.088335,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,1743.113941,3657.031569,0.843813,0.283814,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,60.000000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,100.000000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,118000.000000,101895.000000,8.000000,1.000000,4.000000


### COLUMN: Surface of the land ###

In [254]:
# checking suspicious surface of the land size
def func_max_land_size():
    max_land_size = df['Surface of the land'].idxmax()
    print(f'index of the property with maximum land size is: {max_land_size}, data: {df.loc[max_land_size, 'Surface of the land']}')

    print(f"{df.loc[max_land_size, 'URL']}")

In [255]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,227.432054,1210.621098,2.744691,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,1743.113941,3657.031569,0.843813,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,60.000000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,100.000000,1030.000000,4.000000,0.000000,2.000000


Found the link:  https://immovlan.be/en/detail/residence/for-sale/6900/marche-en-famenne/vbd16284

Why the the 'surface garden' (118000) is BIGGER than 'total land surface' (101895)?

### COLUMN: Number of facades ###

In [256]:
# checking suspicious number of facades
def func_max_facade_number():
    max_facade_number = df['Number of facades'].idxmax()
    print(f'index of the property with maximum number of facades is: {max_facade_number}, data: {df.loc[max_facade_number, 'Number of facades']}')

    print(f"{df.loc[max_facade_number, 'URL']}")

In [257]:
func_max_facade_number()

index of the property with maximum number of facades is: 7080, data: 8.0
https://immovlan.be/en/detail/residence/for-sale/1000/brussels/vbd93085


In [258]:
# Cleaned the property facades
df.loc[7080, 'Number of facades'] = 2


In [259]:
df.loc[7080, 'Number of facades']

np.float64(2.0)

In [260]:
df.describe(include='all')

,URL,Zip code,City,Type of property,Subtype of property,Price,Type of sale,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973,12973.000000,12973,12973,12973,1.297300e+04,12973,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,6086.000000,10172.000000,4415.000000,10219.000000
unique,12973,NaN,1417,2,15,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,https://immovlan.be/en/detail/residence/for-sa...,NaN,Ukkel,House,residence,NaN,for sale,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,467,6902,5811,NaN,12855,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,3860.137593,NaN,NaN,NaN,4.361525e+05,NaN,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,227.432054,1210.621098,2.744101,0.088335,1.610236
std,NaN,2851.473265,NaN,NaN,NaN,5.259908e+05,NaN,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,1743.113941,3657.031569,0.842235,0.283814,1.396896
min,NaN,1000.000000,NaN,NaN,NaN,6.200000e+02,NaN,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,0.000000,0.000000
25%,NaN,1180.000000,NaN,NaN,NaN,2.300000e+05,NaN,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,224.000000,2.000000,0.000000,1.000000
50%,NaN,3071.000000,NaN,NaN,NaN,3.338530e+05,NaN,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,60.000000,494.000000,2.000000,0.000000,1.000000
75%,NaN,6200.000000,NaN,NaN,NaN,4.890000e+05,NaN,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,100.000000,1030.000000,4.000000,0.000000,2.000000


### COLUMN:  Swimming pool ###

In [261]:
# Filling Nan with 0
df['Swimming pool'] = df['Swimming pool'].fillna(0) 

In [262]:
df.describe()

,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,6086.000000,10172.000000,12973.000000,10219.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,227.432054,1210.621098,2.744101,0.030062,1.610236
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,1743.113941,3657.031569,0.842235,0.170766,1.396896
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,60.000000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,100.000000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,118000.000000,101895.000000,4.000000,1.000000,4.000000


In [263]:
df['Swimming pool'].value_counts()

Swimming pool
0.0    12583
1.0      390
Name: count, dtype: int64

In [264]:
df['Swimming pool'].value_counts(normalize=True) * 100

Swimming pool
0.0    96.993756
1.0     3.006244
Name: proportion, dtype: float64

### COLUMN: State of the building ###

In [265]:
n_state_of_the_building_missing = df['State of the building'].isnull().sum()
print(f"Number of missing data on state of the building: {n_state_of_the_building_missing}")

state_of_the_building_mode = df['State of the building'].mode()[0]
print(f"The state of the building 'mode' is: {state_of_the_building_mode}")

state_of_the_building_mean_old = df['State of the building'].mean()
print(f"The state of the building 'mean' BEFORE fillna: {state_of_the_building_mean_old}")

# filling empty cells with 1 as the most obvious choice
df['State of the building'] = df['State of the building'].fillna(1)

state_of_the_building_mean_new = df['State of the building'].mean()
print(f"The state of the building 'mean' AFTER fillna: {state_of_the_building_mean_new}")


df.describe()


Number of missing data on state of the building: 2754
The state of the building 'mode' is: 1.0
The state of the building 'mean' BEFORE fillna: 1.6102358352089245
The state of the building 'mean' AFTER fillna: 1.4806906652277807


,Zip code,Price,Number of rooms,Living Area,Fully equipped kitchen,Furnished,Open fire,Terrace,Terrace area,Garden,Garden area,Surface of the land,Number of facades,Swimming pool,State of the building
count,12973.000000,1.297300e+04,12729.000000,12181.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,12973.000000,6086.000000,10172.000000,12973.000000,12973.000000
mean,3860.137593,4.361525e+05,2.946736,163.649782,2.081554,0.030833,0.027596,0.742928,17.032992,0.542049,227.432054,1210.621098,2.744101,0.030062,1.480691
std,2851.473265,5.259908e+05,1.896229,134.782079,0.431896,0.172872,0.163818,0.437036,35.995164,0.498248,1743.113941,3657.031569,0.842235,0.170766,1.264643
min,1000.000000,6.200000e+02,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,1.000000,0.000000,0.000000
25%,1180.000000,2.300000e+05,2.000000,90.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,224.000000,2.000000,0.000000,1.000000
50%,3071.000000,3.338530e+05,3.000000,132.000000,2.000000,0.000000,0.000000,1.000000,19.000000,1.000000,60.000000,494.000000,2.000000,0.000000,1.000000
75%,6200.000000,4.890000e+05,4.000000,194.000000,2.000000,0.000000,0.000000,1.000000,20.000000,1.000000,100.000000,1030.000000,4.000000,0.000000,2.000000
max,9991.000000,2.600000e+07,100.000000,3000.000000,3.000000,1.000000,1.000000,1.000000,2968.000000,1.000000,118000.000000,101895.000000,4.000000,1.000000,4.000000
